### PR TITLE
redesign test for key retrieval via http

### LINKED ISSUE #427: Integration tests for populating keys from a URL are broken
The integration tests for 'File /home/test_user_keys_from_url/.ssh/authorized_keys' are failing.  @majormoses has updated their keys on GitHub, and so the integration tests that validated that they keys were properly downloaded and put in `authorized_keys` are no longer matching the key content.  We should update the keys in `default_spec.rb`, or rewrite the tests as suggested here: https://github.com/chef-cookbooks/users/blob/63bd82aaec10be9ac586ab04d34de5b4c3278689/test/integration/default/default_spec.rb#L25-L27

### PR DESCRIPTION
**Pull request by karmix**
### Description

Revises test cookbook so that instead of downloading an individual's personal keys from GitHub, it
starts a web service and uses it to serve dedicated keys for validating that the `user_manage` resource can retrieve ssh keys via http.

### Issues Resolved

Resolves #427 

### Check List

- [x] All tests pass.  **Except group tests on opensuse, which are failing for unrelated reasons.**
  See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by karmix** [comment]
It looks like the travis-ci build is having problems installing ChefDK while setting up the test environment.  This error makes it look like it might have a corrupt `apt-get` cache, or be hitting a broken mirror: https://travis-ci.org/chef-cookbooks/users/jobs/540010073#L574 .

File: resources/manage.rb [hunk 1]
```diff
@@ -91,7 +91,7 @@
       ssh_keys = []
       if u['ssh_keys']
         Array(u['ssh_keys']).each do |key|
-          if key.start_with?('https')
+          if key.start_with?('http')
```
Comment (majormoses) [inline resources/manage.rb]: we talked about this at the sous-chefs meeting on Jul 22 2019, the concern is MITM and if we want to support use cases pulling over http I think it needs to be a `curl --insecure` type setup.

**Comment by karmix** [comment]
Sorry, I thought I wrapped this up long ago, but apparently I didn't update the PR.  I added a prefix to non-ssl http, so it's now `INSECURE:http://...`.  `kitchen verify` passed, and there was nothing unexpected with the rebase.

However, it appears that recent changes set `enforce_idempotency` in kitchen.  Because there is no way to ignore specific resources with `enforce_idempotency`, and the `users_manage` resource does not support `:before` notifications, I don't see a portable way to make this patch start the web service for the `users_manage` resource without tripping up the idempotency checks.

**Comment by majormoses** [comment]
Hmm I will try to dig into that issue if I can find some time this weekend. I imagine there has to be a way to give a list of resources that are expected to lack `idempotency` because of their nature either through configuration initialization or inline. I have not been deeply involved in the chef world for a while so maybe I am remembering things incorrectly.

**Review by xorima — APPROVED** [review]

**Review by xorima — APPROVED** [review]

**Review by xorima — APPROVED** [review]

**Review by xorima — APPROVED** [review]

**Comment by Stromweld** [comment]
closing this due to age. please feel free to reopen if needed.

**Pull request closed without merging**